src/mgr/MetricCollector.cc

```diff
@@ -114,6 +114,36 @@ void MetricCollector<Query, Limit, Key, Report>::remove_all_queries() {
   }
 }
 
+template <typename Query, typename Limit, typename Key, typename Report>
+int MetricCollector<Query, Limit, Key, Report>::reregister_query(MetricQueryID query_id) {
```

Comment:
You can do away with the query id since this is basically a brute-force register.